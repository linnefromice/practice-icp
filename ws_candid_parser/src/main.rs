use std::path::Path;

use anyhow::Result;
use candid::pretty_check_file;

fn main() -> Result<()> {
    Ok(())
}

#[cfg(test)]
mod tests {
    use super::*;

    use insta::assert_snapshot;

    #[test]
    fn test_env_from_file() {
        let candid_path = "assets/sample.did";
        let (env, _) = pretty_check_file(Path::new(candid_path)).unwrap();
        let config = candid::bindings::rust::Config::new();
        let result = candid::bindings::rust::compile(&config, &env, &None);
        assert_snapshot!(result);
    }

    #[test]
    fn test_actor_from_file() {
        let candid_path = "assets/sample.did";
        let (env, actor) = pretty_check_file(Path::new(candid_path)).unwrap();
        let config = candid::bindings::rust::Config::new();
        let result = candid::bindings::rust::compile(&config, &env, &Some(actor.unwrap()));
        assert_snapshot!(result);
    }
}