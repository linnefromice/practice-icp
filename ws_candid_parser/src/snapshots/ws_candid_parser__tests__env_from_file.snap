---
source: src/main.rs
expression: result
---
// This is an experimental feature to generate Rust binding from Candid.
// You may want to manually adjust some of the types.
#![allow(dead_code, unused_imports)]
use candid::{self, CandidType, Deserialize, Principal, Encode, Decode};
use ic_cdk::api::call::CallResult as Result;

#[derive(CandidType, Deserialize)]
pub struct Body { pub data: Box<BodyData> }

#[derive(CandidType, Deserialize)]
pub struct BodyData { pub data: Box<CurrencyData> }

#[derive(CandidType, Deserialize)]
pub struct CanisterMetricsSnapshot {
  pub cycles: candid::Nat,
  pub timestamp: u64,
}

#[derive(CandidType, Deserialize)]
pub struct CurrencyData { pub quote: Box<Quote> }

#[derive(CandidType, Deserialize)]
pub struct Data { pub body: Body }

#[derive(CandidType, Deserialize)]
pub enum Env { Production, Test, LocalDevelopment }

#[derive(CandidType, Deserialize)]
pub enum InitError {
  InvalidDestination(String),
  InvalidPrincipal(Principal),
  InvalidContent(String),
  InvalidRequest(String),
}

#[derive(CandidType, Deserialize)]
pub struct Quote { pub usd: Box<QuoteData> }

#[derive(CandidType, Deserialize)]
pub struct QuoteData { pub volume_24h: f64, pub price: f64 }

#[derive(CandidType, Deserialize)]
pub enum Result_ { Ok, Err(InitError) }

#[derive(CandidType, Deserialize)]
pub struct Snapshot { pub value: Box<SnapshotValue>, pub timestamp: u64 }

#[derive(CandidType, Deserialize)]
pub struct SnapshotValue { pub data: Data }


